test/NuGet.Frameworks.Test/FrameworkReducerTests.cs

```diff
@@ -11,6 +12,74 @@ namespace NuGet.Test
+        [InlineData("uap10.0", "dotnet,portable-win81+net45", "dotnet")]
+        [InlineData("uap10.0", "wpa81,dotnet5.2,portable-win81+net45", "wpa81")]
+        [InlineData("uap10.0", "dotnet5.1,portable-win81+net45", "dotnet5.1")]
+        [InlineData("uap10.0", "wpa81,dotnet5.1,portable-win81+net45", "wpa81")]
+        [InlineData("uap10.0", "dotnet5.1,portable-win81+net45", "dotnet5.1")]
+        // Take the most specific PCL profile
+        [InlineData("uap10.0", "dotnet6.0,portable-win81+net45", "portable-win81+net45")]
+        [InlineData("uap10.0", "dotnet6.0,portable-win81+net45+sl5,portable-win81+net45", "portable-win81+net45")]
+        [InlineData("uap10.0", "dotnet6.0,portable-win81+net45+sl5,portable-uap11.0", "portable-win81+net45+sl5")]
+        // Same TFM wins
+        [InlineData("net461", "net462,net461,net46,net45,net4,net2,dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "net461")]
+        [InlineData("net461", "net46,net45,net4,net2,dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "net46")]
+        [InlineData("net461", "net45,net4,net2,dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "net45")]
+        [InlineData("net461", "net4,net2,dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "net4")]
+        [InlineData("net461", "net2,dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "net2")]
+        // Use a compatible TFM if there are no framework matches
+        [InlineData("net461", "dotnet6.0,dotnet5.4,dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "dotnet5.4")]
+        [InlineData("net461", "dotnet5.3,dotnet,portable-net45+win8,portable-net45+win8+wpa81", "dotnet5.3")]
+        [InlineData("net461", "dotnet,portable-net45+win8,portable-net45+win8+wpa81", "dotnet")]
+        [InlineData("net461", "portable-net45+win8,portable-net45+win8+wpa81", "portable-net45+win8")]
+        [InlineData("net461", "portable-net45+win8+wpa81,native", "portable-net45+win8+wpa81")]
+        [InlineData("net7", "dotnet6.0,dotnet5.4,portable-net45+win8", "dotnet5.4")]
+        // Additional tests
+        [InlineData("dotnet5.5", "dotnet6.0,dotnet5.4,portable-net45+win8", "dotnet5.4")]
+        [InlineData("dotnet7", "dotnet6.0,dotnet5.4,portable-net45+win8", "dotnet6.0")]
+        [InlineData("dnxcore50", "dotnet6.0,dotnet5.3,portable-net45+win8", "dotnet5.3")]
+        [InlineData("dotnet", "dotnet5.1,native,uap10.1", null)]
+        [InlineData("uap10.0", "uap10.1,dotnet5.4,dotnet5.3.0.1,dotnet6.0,dnxcore50,native", null)]
+        [InlineData("dotnet", "", null)]
+        [InlineData("dotnet5.4", "", null)]
+        [InlineData("uap10.0", "", null)]
+        [InlineData("net461", "", null)]
+        public void FrameworkReducer_GetNearestWithGenerations(
```

Comment:
This is an excellent test :). Really helps visualize the exact behavior. Awesome!